### PR TITLE
Add support for upsample_nearest2d op in the Arm backend

### PR DESCRIPTION
**Pull request by Tessil**
Add support for the `torch.ops.aten.upsample_nearest2d.vec` operator in the Arm backend.

The `torch.ops.aten.upsample_nearest2d.vec` operator is added to the operators to not decompose so that it can directly be legalized to a `tosa.resize` operator. 

Change-Id: Id0b742214e5432957b2f573b4218f09a4d9734e4

### CODE REVIEW

File: backends/arm/test/ops/test_upsample_nearest2d.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from typing import Optional, Tuple
+
+import torch
+from executorch.backends.arm.test import common
+from executorch.backends.arm.test.tester.arm_tester import ArmTester
+from parameterized import parameterized
+
+
+test_data_suite = [
+    # (test_name, test_data, size, scale_factor, compare_outputs)
```
Comment (digantdesai) [inline backends/arm/test/ops/test_upsample_nearest2d.py:18]: ❤️ these test. Awesome!

File: backends/arm/test/tester/arm_tester.py [hunk 2]
```diff
@@ -34,53 +34,65 @@
...
 from executorch.exir.backend.compile_spec_schema import CompileSpec
+from executorch.exir.backend.partitioner import Partitioner
 from executorch.exir.lowered_backend_module import LoweredBackendModule
 from torch.fx import Graph
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
+def _dump_lowered_modules_artifact(
```
Comment (digantdesai) [inline backends/arm/test/tester/arm_tester.py:45]: Just refactoring I assume? Try to keep these in a different commit (same PR is fine).

**Review by digantdesai — APPROVED** [review]

File: backends/arm/test/tester/arm_tester.py (hunk 2 above)
Comment (Tessil) [reply backends/arm/test/tester/arm_tester.py:45]: Mainly because I needed the new `class ToEdgeTransformAndLower(tester.ToEdgeTransformAndLower)` and so I extracted this so that it can be shared by `class Partition(tester.Partition)`.

I can eventually create a separate commit for the whole `backends/arm/test/tester/arm_tester.py` changes if you prefer as it's effectively not directly related to the upsample_nearest2d. Just some additions to the tester to properly test the op.

Comment (digantdesai) [reply backends/arm/test/tester/arm_tester.py:45]: All good. We can do different commits next time :)

**Comment by Tessil** [comment]
Thanks for the review. I rebased the change and fixed the conflicts.

**Comment by digantdesai** [comment]
Internal checker found these type issues, can you please fix and rebase? Thanks.
```
Found type errors!
executorch/backends/arm/tosa_utils.py:356:16 Undefined or invalid type [11]: Annotation `torch.tensor` is not defined as a type.
executorch/backends/arm/tosa_utils.py:373:21 Incompatible parameter type [6]: In call `torch._C._VariableFunctions.all`, for 1st positional argument, expected `Tensor` but got `bool`.
executorch/backends/arm/tosa_utils.py:374:21 Incompatible parameter type [6]: In call `torch._C._VariableFunctions.all`, for 1st positional argument, expected `Tensor` but got `bool`.
```

**Comment by Tessil** [comment]
Thanks, the last change should fix this.

**Comment by digantdesai** [comment]
still seems like a type errors on op_upsample_nearest2d.py:48 get_resize_parameters for 3rd arg expected `ResizeMode` but got `int`. Sorry for the delay.

**Comment by digantdesai** [comment]
Also rebase?

**Comment by Tessil** [comment]
Thanks, I rebased and fixed the issue.

**Comment by digantdesai** [comment]
Sorry rebase?

**Comment by Tessil** [comment]
@digantdesai Thanks, done

**Comment by digantdesai** [comment]
If CI is not green, please make it a Draft PR

**Comment by Tessil** [comment]
The failure doesn't seem to be related to this PR as it's the new `test_var.py` tests introduced during the rebase from upstream that are failing. I'll check internally if these can be fixed in another PR.

**Comment by digantdesai** [comment]
Rebase please?

**Comment by freddan80** [comment]
Pls fix the failing unit test.

**Comment by Tessil** [comment]
The unit test failing is not due to the code of this PR, we confirmed it internally. We should check to raise another PR to fix the `torch.var` test failure as it's not directly related to this one.

**Comment by freddan80** [comment]
Rebase please

**Comment by Tessil** [comment]
With https://github.com/pytorch/executorch/commit/995c2bf0392ac3843e5e255ac5d06538458d0b2c fixing the test_var accuracy issue, the tests are now passing.

**Review by freddan80 — APPROVED** [review]

**Comment by freddan80** [comment]
@digantdesai I'm blocked from merging dues to
"Meta Internal-Only Changes Check: This Pull Request has changes that have not been imported to the Diff, please import the Pull Request"

Do you mind helping out?

**Comment by digantdesai** [comment]
Sigh, sorry can you please rebase?
